File: frontend/src/types/applicationTemplate/CvConfigurationDto.ts
import { KeyValueObject } from '@/types/KeyValueObject.ts'

export interface CvConfigurationDto {
  includedCvContent: {
    includedWorkExperience: CvEntrySelectionDto[]
    includedEducation: CvEntrySelectionDto[]
    includedProjects: CvEntrySelectionDto[]
    includedSkills: number[]
  } | undefined
  cvStyle: string
  cvStyleOptions: KeyValueObject<string>
}

export interface CvEntrySelectionDto {
  id: number
  includeDescription: boolean
}

File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/applicationTemplate/dto/CvConfigurationDto.kt
package ch.streckeisen.mycv.backend.applicationTemplate.dto

data class CvConfigurationDto(
    val cvStyle: String,
    val includedCvContent: IncludedCvContentDto?,
    val cvStyleOptions: Map<String, String>?
)

data class IncludedCvContentDto(
    val includedWorkExperience: List<CvEntrySelectionDto>,
    val includedEducation: List<CvEntrySelectionDto>,
    val includedProjects: List<CvEntrySelectionDto>,
    val includedSkills: List<Long>,
)

data class CvEntrySelectionDto(
    val id: Long,
    val includeDescription: Boolean
)
